packages/venia-concept/src/actions/cart/asyncActions.js

```diff
@@ -119,11 +118,11 @@ export const addItemToCart = (payload = {}) => {
 
             // check if the guest cart has expired
             if (noGuestCartId || (response && response.status === 404)) {
-                // if so, then delete the cached ID...
-                // in contrast to the save, make sure storage deletion is
+                // Delete the cached ID from local storage and Redux.
+                // In contrast to the save, make sure storage deletion is
                 // complete before dispatching the error--you don't want an
                 // upstream action to try and reuse the known-bad ID.
-                await clearGuestCartId();
+                await dispatch(removeCart());
```

Comment:
`removeCart` is new - it calls `clearGuestCartId` AND `actions.reset` to remove the `guestCartId` from redux.